pkg/nsx/services/realizestate/realize_state.go

```diff
@@ -55,6 +65,9 @@ func (service *RealizeStateService) CheckRealizeState(backoff wait.Backoff, inte
 			if *result.State == model.GenericPolicyRealizedResource_STATE_REALIZED {
 				return nil
 			}
+			if *result.State == model.GenericPolicyRealizedResource_STATE_ERROR {
+				return NewRealizeStateError(fmt.Sprintf("%s realized with error", entityType))
```

Comment:
Can we also put any reason in result to the error massage?

Reply:
Added, thanks!